### ISSUE TITLE
current nightly dotty is crashed with sbt projec

### ISSUE THREAD

**Issue by rssh** [lampepfl/dotty#10772]
## Minimized code

create a new project
```
sbt new lampepfl/dotty.g8 
```

and change in build.sbt
```
val dottyVersion = "3.0.0-M2"
```
to
```
val dottyVersion = dottyLatestNightlyBuild.get
```

Then try to compile project.

Other versions:
JDK8,  JDK 15,   
sbt-1.4.4,  sbt-dotty-1.4.6

## Output (click arrow to expand)
<details>

```
[info] compiling 1 Scala source to /Users/rssh/tests/dotty/test/dotty-project-template/target/scala-3.0.0-M3/classes ...
java.lang.LinkageError: loader constraint violation: loader dotty.tools.sbtplugin.DottyPlugin$DottyLoader$1 @73565db4 wants to load interface xsbti.AnalysisCallback. A different interface with the same name was previously loaded by sbt.internal.MetaBuildLoader @4b44655e. (xsbti.AnalysisCallback is in unnamed module of loader sbt.internal.MetaBuildLoader @4b44655e, parent loader sbt.internal.FullScalaLoader @7ea37dbf) while compiling /Users/rssh/tests/dotty/test/dotty-project-template/src/main/scala/Main.scala
[error] ## Exception when compiling 1 sources to /Users/rssh/tests/dotty/test/dotty-project-template/target/scala-3.0.0-M3/classes
[error] java.lang.LinkageError: loader constraint violation: loader dotty.tools.sbtplugin.DottyPlugin$DottyLoader$1 @73565db4 wants to load interface xsbti.AnalysisCallback. A different interface with the same name was previously loaded by sbt.internal.MetaBuildLoader @4b44655e. (xsbti.AnalysisCallback is in unnamed module of loader sbt.internal.MetaBuildLoader @4b44655e, parent loader sbt.internal.FullScalaLoader @7ea37dbf)
[error] java.base/java.lang.ClassLoader.defineClass1(Native Method)
[error] java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1016)
[error] java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:151)
[error] java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:514)
[error] java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:422)
[error] java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:416)
[error] java.base/java.security.AccessController.doPrivileged(AccessController.java:691)
[error] java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:415)
[error] java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
[error] java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
[error] dotty.tools.dotc.core.Contexts$Context.sbtCallback(Contexts.scala:200)
[error] dotty.tools.dotc.sbt.ExtractDependencies.isRunnable(ExtractDependencies.scala:55)
[error] dotty.tools.dotc.Run.runPhases$4$$anonfun$4(Run.scala:181)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:15)
[error] scala.runtime.function.JProcedure1.apply(JProcedure1.java:10)
[error] scala.collection.ArrayOps$.foreach$extension(ArrayOps.scala:1323)
[error] dotty.tools.dotc.Run.runPhases$5(Run.scala:195)
[error] dotty.tools.dotc.Run.compileUnits$$anonfun$1(Run.scala:203)
[error] scala.runtime.function.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
[error] dotty.tools.dotc.util.Stats$.maybeMonitored(Stats.scala:67)
[error] dotty.tools.dotc.Run.compileUnits(Run.scala:210)
[error] dotty.tools.dotc.Run.compileSources(Run.scala:146)
[error] dotty.tools.dotc.Run.compile(Run.scala:130)
[error] dotty.tools.dotc.Driver.doCompile(Driver.scala:39)
[error] dotty.tools.xsbt.CompilerBridgeDriver.run(CompilerBridgeDriver.java:88)
[error] dotty.tools.xsbt.CompilerBridge.run(CompilerBridge.java:22)
[error] sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:92)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:186)
[error] scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[error] sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:241)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:176)
[error] sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:157)
[error] sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:239)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:157)
[error] sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:204)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:573)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:573)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:173)
[error] sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:171)
[error] sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:458)
[error] sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:116)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[error] sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[error] sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:261)
[error] sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$8(Incremental.scala:413)
[error] sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:498)
[error] sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:400)
[error] sbt.internal.inc.Incremental$.apply(Incremental.scala:165)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:573)
[error] sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:491)
[error] sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[error] sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:420)
[error] sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)

```
</details>

**Comment by smarter** [comment]
It's because we need to release a new version of sbt-dotty after https://github.com/lampepfl/dotty/pull/10607 but I'm waiting for https://github.com/lampepfl/dotty/pull/10762 which also changes sbt-dotty to do that /cc @abgruszecki

**Comment by smarter** [comment]
Fixed by upgrading to sbt-dotty 0.5.0
